File: 1-Semestre/Desarrollo de algoritmos/Actividades/Diagramas de flujo-Python (Semana 5)/Ejercicio 5.py
'''
Determinar el porcentaje de varones y de mujeres que hay en un salón de clases.
Todo número requerido para realizar el cálculo es ingresado por el usuario.
'''

print("Este programa calcula el porcentaje de hombres y mujeres en un salón de clases.")
muj = int(input("Ingrese la cantidad de mujeres: "))
hom = int(input("Ingrese la cantidad de hombres: "))

while muj <= 0 or hom <= 0: # Comprueba que los valores sean mayores a 0
    print("[!] La cantidad de hombres y mujeres debe ser mayor a 0.")
    muj = int(input("Ingrese la cantidad de mujeres: "))
    hom = int(input("Ingrese la cantidad de hombres: "))

total = hom + muj # Total de alumnos
p_muj = (muj / total) * 100 # Porcentaje de mujeres
p_hom = (hom / total) * 100 # Porcentaje de hombres

print(f"\nDe un total de {total} alumnos hay {p_muj:.2f}% de mujeres y {p_hom:.2f}% de hombres en el salón.") ## .2f para redondear a 2 decimales

File: 2-Semestre/Taller de Programación/Personal/Python-C#/Ejercicio 5.cs
/*
Determinar el porcentaje de varones y de mujeres que hay en un salón de clases.
Todo número requerido para realizar el cálculo es ingresado por el usuario.
*/

using System;

namespace Ejercicio_5
{
    public class Program
    {
        public static void Main(string[] args)
        {
            Console.WriteLine("Este programa calcula el porcentaje de hombres y mujeres en un salón de clases.\n");

            int mujeres, hombres;

            do
            {
                Console.Write("Ingrese la cantidad de mujeres: ");
                mujeres = Convert.ToInt32(Console.ReadLine());

                if (mujeres <= 0)
                {
                    Console.WriteLine("[!] La cantidad de mujeres no debe ser menor o igual a 0");
                }
            } while (mujeres <= 0);

            do
            {
                Console.Write("Ingrese la cantidad de hombres: ");
                hombres = Convert.ToInt32(Console.ReadLine());

                if (hombres <= 0)
                {
                    Console.WriteLine("[!] La cantidad de hombres no debe ser menor o igual a 0");
                }
            } while (hombres <= 0);


            double total = mujeres + hombres; // Total de alumnos
            double porcentaje_mujeres = (mujeres / total) * 100; // Porcentaje de mujeres
            double porcentaje_hombres = (hombres / total) * 100; // Porcentaje de hombres

            Console.WriteLine($"\nDe un total de {total} alumnos hay {porcentaje_mujeres}% de mujeres y {porcentaje_hombres}% de hombres en el salón.");
        }
    }
}